Added `example/shapenet.py`

taiya opened the pull request.

Commit:
shapenet example

Qwlouse commented on examples/shapenet.py:
```diff
@@ -0,0 +1,97 @@
+# --- make object flat on X/Y and not penetrate floor
+obj.quaternion = kb.Quaternion(axis=[1,0,0], degrees=90)
+# HACK: bounds are not updated after rotation! supposed to be obj.bounds[0][2]
+obj.position = obj.position - (0, 0, obj.bounds[0][1])  
```
```suggestion
obj.position = obj.position - (0, 0, obj.aabbox[0][1])  
```

Qwlouse replied:
The bounds are pre-transformation, here you want the aabbox or bbox_3d property instead.

Qwlouse commented on examples/shapenet.py:
```diff
@@ -0,0 +1,97 @@
+  seed=1,
+  frame_start=1,
+  frame_end=30,
+  resolution=(256, 256),
```
this does nothing, instead set width and height

Qwlouse commented on kubric/__init__.py:
```diff
@@ -47,15 +47,9 @@
-
 from kubric.custom_types import AddAssetFunction
 from kubric.custom_types import PathLike
+from kubric.custom_types import Quaternion
```
```suggestion
from pyquat import Quaternion
```

taiya replied:
The point of this is to make it clear what's the rotational type used within the library?

Qwlouse commented on kubric/custom_types.py:
```diff
@@ -25,3 +26,5 @@
 
 ArrayLike = Union[Sequence[float], np.ndarray]
+
+Quaternion = pyquat.Quaternion
```
```suggestion
```

custom_types is meant more as a place for type-hinting  types

taiya replied:
in that case, should we have another "types" module, where all the default types are specified?

Qwlouse commented on kubric/custom_types.py:
```diff
@@ -16,6 +16,7 @@
 from typing import Any, Callable, Union, Sequence
 
 import numpy as np
+import pyquaternion as pyquat
```
```suggestion
```

Commit:
Update kubric/__init__.py

Co-authored-by: Klaus Greff <klausg@google.com>

Commit:
Update kubric/custom_types.py

Co-authored-by: Klaus Greff <klausg@google.com>

Commit:
bounds → aabox

Co-authored-by: Klaus Greff <klausg@google.com>

Commit:
addressed Klaus' PR review

Commit:
Merge branch 'example/shapenet' of github.com:google-research/kubric into example/shapenet

Commit:
pytlint

Commit:
pylint fix (+updated KubruntuDev)

running `make pylint` gave no errors, but the github action was giving some, despite using the same .pylintrc

this commit fixes the problem by using the dev container to run linting targets, which re
also now used by `make pylint`

Commit:
pylint fixes (kubruntudev)

Commit:
improved makefile

Commit:
fixed aabox
